src/test/unit/io/stan_csv_reader_test.cpp

```diff
@@ -68,7 +68,7 @@ TEST_F(StanIoStanCsvReader,read_metadata1) {
 }
 TEST_F(StanIoStanCsvReader,read_header1) {
   Eigen::Matrix<std::string, Eigen::Dynamic, 1> header;
-  EXPECT_TRUE(stan::io::stan_csv_reader::read_header(header1_stream, header));
+  EXPECT_TRUE(stan::io::stan_csv_reader::read_header(header1_stream, header, 0));
```

Comment:
With 0 passed in, nothing ever gets written, so that branch doesn't get tested.

Add the exact same test with a copy of the header stream and a stringstream ptr for output.

Extra credit for testing the output strings.